app/assets/javascripts/pooled_tubes_from_whole_plates.js

```diff
@@ -103,50 +113,77 @@
+          this.labwareContainer().addClass('scan-plate')
           pooler.checkLabwares()
         },
         badLabware: function () {
           this.clearLabware()
-          this.plateContainer().removeClass('good-plate wait-plate scan-plate')
-          this.plateContainer().addClass('bad-plate')
+          this.labwareContainer().removeClass('good-plate wait-plate scan-plate')
+          this.labwareContainer().addClass('bad-plate')
           $('#summary_tab').addClass('ui-disabled')
         },
         goodLabware: function () {
-          this.plateContainer().removeClass('bad-plate wait-plate scan-plate')
-          this.plateContainer().addClass('good-plate')
+          this.labwareContainer().removeClass('bad-plate wait-plate scan-plate')
+          this.labwareContainer().addClass('good-plate')
           pooler.checkLabwares()
         },
-        plateContainer: function () {
+        labwareContainer: function () {
           return $(this).closest('.plate-container')
         },
-        checkLabware: function (data, status) {
+        checkLabware: function (data, status, scanned_barcode) {
           var response = data[this.dataset.labwareType]
           if (SOURCE_STATES.indexOf(response.state) === -1) {
             this.badLabware()
             SCAPE.message('Scanned ' + this.dataset.labwareType + 's are unsuitable', 'invalid')
           } else {
-            if (pooler.record(response, $(this).data('position'))) {
+            var position = $(this).data('position')
+            if (pooler.record(response, position, scanned_barcode)) {
               this.goodLabware()
             } else {
+              var clashing_labwares = []
+
+              Object.keys(pooler.labware_details).forEach(function (key) {
+                var current_labware_details = pooler.labware_details[key]
+                // skip empty labware locations
+                if (current_labware_details.tags == undefined) {
+                  return
+                }
+
+                // check for clashing tags
+                if (current_labware_details.tags.includes(pooler.clashing_tags)) {
+                  var human_barcode = String(current_labware_details.human_barcode)
+                  var machine_barcode = String(current_labware_details.machine_barcode)
+                  clashing_labwares.push('' + human_barcode + ' (' + machine_barcode + ')')
+                }
+              })
+
+              var clashing_labwares_string = clashing_labwares.join(', ')
+
               this.badLabware()
-              SCAPE.message('Scanned ' + this.dataset.labwareType + 's have matching tags', 'invalid')
+
+              SCAPE.message(
+                'The scanned ' +
+                  this.dataset.labwareType +
+                  ' contains tags that would clash with those in other ' +
+                  this.dataset.labwareType +
+                  's in the pool. Tag clashes found between: ' +
+                  clashing_labwares_string,
```

Comment:
Might it be worth concatenating this message in a local variable prior to calling this function?  Then the `'invalid'` as a second argument will be more noticeable to people unfamiliar to this code.